app/src/main/java/unipd/se18/ocrcamera/TextExtractor.java

```diff
@@ -0,0 +1,240 @@
+
+    /**
+     * Index of the point on the bottom-right position.
+     * More details at:
+     * {@link FirebaseVisionText.TextBlock#getCornerPoints()},
+     * {@link FirebaseVisionText.Line#getCornerPoints()}
+     * or {@link FirebaseVisionText.Element#getCornerPoints()}.
+     */
+    private final int BOTTOM_RIGHT = 3;
+
+    /*
+    The next method is required by the OCRInterface that avoid a single point of failure.
+     */
+
+    /**
+     * Extracts a text from a given image.
+     * More details at: {@link FirebaseVisionText}.
+     * @param img The image in a Bitmap format
+     * @return The String of the ingredients recognized, empty String if nothing is recognized
+     * @author Pietro Prandini (g2)
+     */
+    public String getTextFromImg(Bitmap img) {
+        // String used for the logs of this method
+        final String methodTag = "getTextFromImg -> ";
+        Log.d(TAG, methodTag + "launched");
+
+        // Extracts the text from the pic
+        FirebaseVisionText firebaseVisionTextExtracted = extractFireBaseVisionText(img);
+
+        // Returns the result
+        if (firebaseVisionTextExtracted != null) {
+            // Text found in img -> returns the String of the text found
+            Log.v(TAG, methodTag + "Text found");
+            return extractString(firebaseVisionTextExtracted);
+        } else {
+            // Text not found in img -> returns an empty string
+            Log.v(TAG, methodTag + "Text not found");
+            return "";
+        }
+    }
+
+    /**
+     * Extracts a FirebaseVisionText from a given image.
+     * More details at: {@link FirebaseVisionText}, {@link CountDownLatch},
+     * {@link Task#addOnSuccessListener(OnSuccessListener)}, {@link OnSuccessListener}.
+     * @param img The image in a Bitmap format
+     * @return The FirebaseVisionText of the text recognized, null if nothing is recognized
+     * @author Pietro Prandini (g2), Luca Moroldo (g3)
+     */
+    private FirebaseVisionText extractFireBaseVisionText(Bitmap img) {
+        // String used for the logs of this method
+        final String methodTag = "extractFireBaseVisionText -> ";
+        Log.d(TAG, methodTag + "launched");
+
+        // Starts the time counter - useful for tests
+        long beforeWaiting = java.lang.System.currentTimeMillis();
+
+        // Settings the image to analyze
+        FirebaseVisionImage firebaseVisionImage = FirebaseVisionImage.fromBitmap(img);
+
+        // Settings the "on device" analyzing method
+        FirebaseVisionTextRecognizer textRecognizer =
+                FirebaseVision.getInstance().getOnDeviceTextRecognizer();
+
+        /*
+        The standard java Semaphore is not used because it's not work properly
+        with the Task<FirebaseVisionText>.
+         */
+
+        // Instantiates the CountDownLatch used for synchronizing the extraction
+        final CountDownLatch extraction = new CountDownLatch(1); // Luca Moroldo (g3)
```

Comment:
I know you did not write this line of code.
However I advise you not to use magic numbers, but create a variable [x = 1] and use this variable instead of the number [CountDownLatch (x)]

Reply:
Thanks, I'll firstly put that number in a variable as you suggested.